### DISCUSSION TITLE
About   codegen

### DISCUSSION THREAD

**Discussion by xbh-sdju** [OpenRealTimeSimulation/SolverCodegen#7] [Q&A] [answered]
Why does it crash when I open the codegen.exe file.

**Comment by OpenRealTimeSimulation** [comment]
Hello @xbh-sdju,

To help you, please elaborate on what you were trying to do leading up to the issue with the codegen executable and provide any error message that you may have received.  In what way did the executable crash?

Also, please provide any information about your computer setup, such as CPU, RAM, and operating system.  This info can help to narrow down the source of the problem.

**Reply by xbh-sdju** [reply]
When I open codegen.exe it crashes and does nothing, no matter how many times I do something like this.

CPU AMD R5 4600H
RAM 16GB
Operating system win11 64 bit

**Reply by OpenRealTimeSimulation** [reply]
When trying to run the codegen executable, are you simply double-clicking on it in the file explorer?  The executable is designed to run within a command prompt and will close immediately if not run from the prompt; its a CLI tool.  Try running codegen.exe from your command prompt (cmd.exe or powershell) and see if it runs there.

The common usage of the codegen tool in the command window is:
`codegen netlist_file`
where _netlist_file_ argument is the name of the text file containing the netlist of the electrical system you want to create a solver for.  The tool will generate a solver C++ source file for the netlist in current directory and immediately return.  If you pass no arguments, the tool will print a summary description of itself.  If you call
`codegen -help`
the tool will print basic help text on how to use the tool.

Please see the user guide for the tool, found either in releases page or in the pinned Getting Started post in discussions page, for more details on how to use the codegen CLI tool and its generated solver code.

**Comment by OpenRealTimeSimulation** [comment, marked as answer]
@xbh-sdju,

In powershell, prefix the codegen command with `.\` to let powershell know to look for codegen in the current directory.  So:
`.\codegen`
`.\codegen -help`
`.\codegen netlist_filename`

By default, powershell doesn't search for commands/executables in current directory, so you need the `.\` to get it to do so.

In Windows cmd.exe, you don't usually need the `.\` prefix.

**Reply by xbh-sdju** [reply]
The problem is finally solved. 
Thank you very much.
